worlds/gl/Rules.py

```diff
@@ -42,7 +33,6 @@ def set_rules(world: "GauntletLegendsWorld"):
                     if location.name not in world.disabled_locations:
                         add_rule(
                             world.get_location(location.name),
-                            lambda state, level_id_=level_id >> 4, difficulty=location.difficulty - 1: prog_count(
-                                state, world.player, difficulty_lambda[level_id_][difficulty],
-                            ),
-                        )
+                            lambda state, level_id_=level_id >> 4, difficulty=location.difficulty - 1:
+                            state.prog_items[world.player]["stones"] >= difficulty_lambda[level_id_][difficulty],
```

Comment:
you can still use the state.has() helpers on custom-added items like your `stones` here, not required because it's very technically slower to use the helper, but will cut down on refactors if the underlying counters change in the future
```suggestion
                            state.has("stones", self.player, difficulty_lambda[level_id_][difficulty])
```